src/hooks/useApi.ts

```diff
@@ -0,0 +1,77 @@
+import useSWRImmutable from 'swr/immutable';
+import { Fetcher, SWRResponse, SWRConfiguration } from 'swr';
+
+import { Commodity } from '@/book/entities';
+import * as queries from '@/book/queries';
+import { PriceDB } from '@/book/prices';
+import getUser from '@/lib/getUser';
+
+/**
+ * Wrapper around SWR to make the usable keys explicit and re-usable
+ */
+export default function useApi(k: ApiPaths | null, args?: any): SWRResponse {
+  let f = k !== null ? API[k].f : () => {};
+  const swrArgs = k !== null ? API[k].swrArgs : undefined;
+
+  let key: string | null = k;
+  if (k === '/api/splits/<guid>') {
+    key = k.replace('<guid>', args.guid);
+    f = () => queries.getSplits(args.guid);
```

Comment:
This is quite ugly, at some point would be good to investigate how to make a proper router with proper types...